build.gradle

```diff
@@ -232,20 +232,32 @@ allprojects {
         }
     }
 
-    configurations {
-        all {
-            resolutionStrategy {
-                def cacheHours = isCiBuild ? 0 : 24
-                cacheDynamicVersionsFor cacheHours, 'hours'
-                cacheChangingModulesFor cacheHours, 'hours'
-                eachDependency { DependencyResolveDetails details ->
-                    //specifying a fixed version for all libraries with 'org.gradle' group
-                    if (details.requested.group == 'org.codehaus.groovy') {
-                        details.useVersion groovyVersion
-                    }
-                    if (details.requested.group == "org.spockframework") {
-                        details.useVersion(spockVersion)
-                    }
+    configurations.configureEach {
+        resolutionStrategy {
+            def cacheHours = isCiBuild ? 0 : 24
+            cacheDynamicVersionsFor cacheHours, 'hours'
+            cacheChangingModulesFor cacheHours, 'hours'
+
+            // Align versions of transitive dependencies with the version we are using
+            eachDependency { DependencyResolveDetails details ->
+                def forcedSpringLibraryUpgrades = ['spring-core', 'spring-jcl', 'spring.orm']
+                def forcedTomcatLibraryUpgrades = ['tomcat-jdbc', 'tomcat-juli']
+                if (details.requested.group == 'org.codehaus.groovy') {
+                    details.useVersion(groovyVersion)
+                }
+                if (details.requested.group == 'org.spockframework') {
+                    details.useVersion(spockVersion)
+                }
+                if (details.requested.group == 'org.springframework' && details.requested.name in forcedSpringLibraryUpgrades) {
+                    details.useVersion(springVersion)
+                }
+/*
+                if (details.requested.group == 'org.yaml' && details.requested.name == 'snakeyaml') {
+                    details.useVersion(snakeyamlVersion)
```

Comment:
Shouldn't we remove snakeyaml since it is commented out?

Reply:
Oops! I was testing something and missed to enable this again. I am unsure about snakeyaml. 2.0 is pulled in by other libraries. Will it work to force 2.2?